internal/bgp/frr/frr.go

```diff
@@ -363,11 +365,65 @@ func NewSessionManager(l log.Logger) *sessionManager {
 	reload := func(config *frrConfig) {
 		generateAndReloadConfigFile(config, l)
 	}
+
 	debouncer(reload, res.reloadConfig, debounceTimeout)
 
+	reloadValidator(l)
+
 	return res
 }
 
+func reloadValidator(l log.Logger) {
+	var tickerIntervals = 5 * time.Second // TODO: CHANGE BACK TO 30 SECOUNDS
+	var prevReloadTimeStamp int
+
+	ticker := time.NewTicker(tickerIntervals)
+	go func() {
+		for range ticker.C {
+			validateReload(l, &prevReloadTimeStamp)
+		}
+	}()
+}
+
+var statusFileName = "/etc/frr_reloader/.status"
+
+func validateReload(l log.Logger, prevReloadTimeStamp *int) {
+	bytes, err := os.ReadFile(statusFileName)
+	if err != nil {
+		_ = level.Error(l).Log("op", "reload-validate", "error", err, "cause", "readFile", "fileName", statusFileName)
+		return
+	}
+
+	lastReloadStatus := strings.Fields(string(bytes))
+
+	if len(lastReloadStatus) == 1 {
```

Comment:
Can we make this simpler, and just assume that if the file is not there, it's because we did not create it yet?
That will make the entry point below simpler

Reply:
simpler is better. done